snuba/datasets/processors/transactions_processor.py

```diff
@@ -472,4 +474,11 @@ def process_message(
         # the following operation modifies the event_dict and is therefore *not* order-independent
         self._process_contexts_and_user(processed, event_dict)
 
-        return InsertBatch([processed], None)
+        if event_dict["data"]["received"] is not None:
```

Comment:
In the schema it's defined as a number, and not nullable, so why do we need this check here?

Reply:
I was thinking of merging this PR first (before we make the change to the schema to make `received` a required field), and then making the schema change after?

Reply:
from schema point of view, if it's required and non-nullable, you probably don't need to check it. @lynnagara do you know what will happen if a payload is not schema compatible, will we drop the payload? Do you know what are the processes to ensure the payloads matched the schema?

Reply:
right now, you'll just get a sentry warning if the schema doesn't match. i'd merge the schema change first to validate that if you're unsure.

Reply:
sounds good. Once the payload is always schema compatible, then we probably don't need this extra check.